### PR TITLE
Add Support for Crypto Pending Orders to Show up in Dashboard

### PR DESCRIPTION
**Pull request by SudharakaP**
⚠️ Depends on https://github.com/opencollective/opencollective-api/pull/6378

Related to https://github.com/opencollective/opencollective/issues/4526

Front End PR: https://github.com/opencollective/opencollective-frontend/pull/6725

### CODE REVIEW

File: server/graphql/v2/mutation/OrderMutations.js [hunk 1]
```diff
@@ -89,7 +90,7 @@ const orderMutations = {
         paymentMethod,
         fromCollective: fromCollective && { id: fromCollective.id },
         collective: { id: collective.id },
-        totalAmount: getOrderTotalAmount(order),
+        totalAmount: paymentMethod.type === PAYMENT_METHOD_TYPE.CRYPTO ? 1 : getOrderTotalAmount(order), // insert 1 as a dummy value for the pending order for crypto
```
Comment (znarf) [inline server/graphql/v2/mutation/OrderMutations.js:93]: That should not be needed. Frontend can simply pass a random value. No need to pollute this.

Comment (SudharakaP) [reply server/graphql/v2/mutation/OrderMutations.js]: Makes sense, I've corrected this. Actually I don't think this is needed anymore. 👍🏽

File: server/paymentProviders/thegivingblock/index.js [hunk 2]
```diff
@@ -96,7 +97,7 @@ export const processOrder = async order => {
   const cryptoToFiatFxRate = await getFxRate(order.data.customData.pledgeCurrency, order.currency);
   if (cryptoToFiatFxRate) {
     const totalAmount = Math.round(order.data.customData.pledgeAmount * cryptoToFiatFxRate);
-    await order.update({ totalAmount });
+    await order.update({ totalAmount, status: orderStatus.PENDING });
```
Comment (znarf) [inline server/paymentProviders/thegivingblock/index.js:100]: Update to PENDING before or after or differently, as there is no guarantee to enter this condition at this stage.

File: server/paymentProviders/thegivingblock/index.js [hunk 3]
```diff
@@ -147,9 +148,7 @@ export const confirmOrder = async order => {
     },
   };
 
-  const creditTransaction = await models.Transaction.createFromContributionPayload(transactionPayload);
-
-  return creditTransaction;
+  return await models.Transaction.createFromContributionPayload(transactionPayload);
```
Comment (znarf) [inline server/paymentProviders/thegivingblock/index.js:151]: No need to await.

File: server/paymentProviders/thegivingblock/index.js (hunk 2 above)
Comment (SudharakaP) [reply server/paymentProviders/thegivingblock/index.js]: Ah, yes, I've changed this to update the order status before the `if` statement.

**Review by znarf — APPROVED** [review]
Looks good!

**Pull request merged**